functests/ptp/ptp.go

```diff
@@ -108,4 +96,437 @@ var _ = Describe("ptp", func() {
+					Expect(len(discoveredInterfaces)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP Supported interfaces on slave/master pods"))
+					testFailed := false
+					Expect(discoveredInterfaces).ToNot(ContainElement(defaultRouteInterface), fmt.Sprint("The interfaces discovered incorrectly. Default Pod Interfaces in list"))
+					podLogs, err := getPodLogs(pod)
+					Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+					for _, line := range strings.Split(podLogs, "\n") {
+						if strings.Contains(line, "PTP capable NICs") && strings.Contains(line, defaultRouteInterface) {
+							testFailed = true
+						}
+					}
+					Expect(testFailed).To(BeFalse(), fmt.Sprint("The interfaces discovered incorrectly. Default Pod Interfaces in Pod's logs"))
+				}
+			})
+
+			// 25733
+			It("PTP daemon apply match rule based on nodeLabel", func() {
+				profileSlave := "Profile Name: slave"
+				profileMaster := "Profile Name: grandmaster"
+				for _, pod := range runningPod.Items {
+					podLogs, err := getPodLogs(pod)
+					Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+					if podRole(pod, ptpSlaveNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileSlave),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileMaster),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+				}
+			})
+
+			// 25738
+			It("Slave can sync to master", func() {
+				var masterID string
+				var slaveMasterID string
+				garandmaster := "assuming the grand master role"
+				for _, pod := range ptpRunnungPods {
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						podLogs, err := getPodLogs(pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						Expect(podLogs).Should(ContainSubstring(garandmaster),
+							fmt.Sprintf("Log message \"%s\" not found in pod's log %s", garandmaster, pod.Name))
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "selected local clock") && strings.Contains(line, "as best master") {
+								masterID = strings.Split(line, " ")[4]
+							}
+						}
+					}
+					if podRole(pod, ptpSlaveNodeLabel) {
+						podLogs, err := getPodLogs(pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "new foreign master") {
+								slaveMasterID = strings.Split(line, " ")[6]
+							}
+						}
+					}
+				}
+				Expect(len(masterID)).NotTo(Equal(0))
+				Expect(len(slaveMasterID)).NotTo(Equal(0))
+				Expect(masterID).To(Equal(slaveMasterID[0:len(masterID)]), "Error match MasterID with the SlaveID. Slave connected to another Master")
```

Comment:
Not sure I understand what you are checking here. If slaveMasterID is a substring, use `strings.HasPrefix(`

Reply:
Changed to Expect(slaveMasterID).Should(HavePrefix(masterID)